### DISCUSSION TITLE
Why does Visualizer show only the upper part of the waveform?

### DISCUSSION THREAD

**Discussion by ghost** [vvvar/PeakEater#56] [Q&A] [answered]
I'm not a DSP developer. 
I wonder, why on the Visualizer only the upper part of the waveform is displayed. Or is it a combined view? Or is it for CPU saving?
I mean, the wave has 1 and -1 limit, it pulls the loudspeaker back and forth. Isn't it more correct displaying both sides of it in a clipper?
My question is solely theoretical. I have little idea how things work. But I would like to know more.

![image](https://github.com/vvvar/PeakEater/raw/master/Resources/screenshots/screenshot-clip-visualizer.png)

![image](https://user-images.githubusercontent.com/88319069/215330634-6bc602e9-456e-41ef-a86b-1335b1746bb8.png)

**Comment by vvvar** [comment, marked as answer]
Hey @o771e .
Ah that's because visualiser shows you a magnitude level(in dB's). The way it works is that it takes a chunk of audio buffer, finds highest possible(e.g. peak) value(whether it was phase or anti-phase) and draws it. This type of visualisation usually considered more useful for clippers and limiters because you need to limit a peak, and it's not so important was it in phase or anti-phase. Closest example in this regard could be [FF Pro L](https://www.fabfilter.com/products/pro-l-2-limiter-plug-in).
Perhaps, i could specify more explicitly how Visualizer works in docs, so it would not confuse. Thx for feedback.
